classes/Customer.php

```diff
@@ -338,6 +342,7 @@ public function getByEmail($email, $passwd = null, $ignore_guest = true)
                 $this->{$key} = $value;
             }
         }
+        $this->id_default_group = (int)Configuration::get('PS_CUSTOMER_GROUP');
```

Comment:
Determining if a Customer instance is logged or unidentified should be done outside the Customer class.